types/test/vue-test.ts

```diff
@@ -154,6 +154,45 @@ const FunctionalScopedSlotsComponent = Vue.extend({
   }
 });
 
+declare function assertBoolean<A>(value: string extends A ? never : (A extends boolean ? A : never)): true;
```

Comment:
Calling `assertBoolean(value)` will only compile if `value` is precisely `boolean`, and _not if_ `value` is `boolean | AnotherType`, _nor if_ `value` is `any` (the latter is why `...lue: string extends A ? nev...` is there—it's just a way to throw if `value` is `any`).